providers/aws-ec2/src/test/java/org/jclouds/aws/ec2/features/RouteTableApiMockTest.java

```diff
@@ -0,0 +1,162 @@
+   }
+
+   public void associateRoute() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/associate_route_table.xml");
+      final String associationId = routeTableApi().associateRouteTable(DEFAULT_REGION, "rtb-8c95c0eb", "subnet-6986410e");
+      assertEquals(associationId, "rtbassoc-fb7fed9d", "Failed to associate route");
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=AssociateRouteTable&RouteTableId=rtb-8c95c0eb&SubnetId=subnet-6986410e");
+   }
+
+   public void disassociateRoute() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/disassociate_route_table.xml");
+      final boolean result = routeTableApi().disassociateRouteTable(DEFAULT_REGION, "rtbassoc-fb7fed9d");
+      assertTrue(result, "Failed to disassociate route");
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DisassociateRouteTable&AssociationId=rtbassoc-fb7fed9d");
+   }
+
+   public void createRoute() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/create_route.xml");
+      final boolean created = routeTableApi().createRoute(DEFAULT_REGION, "rtb-a77f2ac0",
+            gatewayId("igw-97e68af3").destinationCidrBlock("172.18.19.0/24"));
+      assertTrue(created, "Failed to match 'true' in test data response");
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION,
+         "Action=CreateRoute&RouteTableId=rtb-a77f2ac0&GatewayId=igw-97e68af3&DestinationCidrBlock=172.18.19.0/24");
+   }
+
+   public void replaceRoute() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/replace_route.xml");
+      final boolean created = routeTableApi().replaceRoute(DEFAULT_REGION, "rtb-a77f2ac0",
+         gatewayId("vgw-1d00376e").destinationCidrBlock("172.18.19.0/24"));
+      assertTrue(created, "Failed to match 'true' in test data response");
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION,
+         "Action=ReplaceRoute&RouteTableId=rtb-a77f2ac0&GatewayId=vgw-1d00376e&DestinationCidrBlock=172.18.19.0/24");
+   }
+
+   public void deleteRoute() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/delete_route.xml");
+      final boolean deleted = routeTableApi().deleteRoute(DEFAULT_REGION, "rtb-a77f2ac0",
+         destinationCidrBlock("172.18.19.0/24"));
+      assertTrue(deleted, "Failed to match 'true' in test data response");
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DeleteRoute&RouteTableId=rtb-a77f2ac0&DestinationCidrBlock=172.18.19.0/24");
+   }
+
+   private RouteTableApi routeTableApi() {
+      return api().getRouteTableApi().get();
+   }
```

Comment:
* Add mock tests that verify the API methods that accept Options objects.
* Add mock tests that enqueue 404 responses for all methods that define such fallbacks, to verify their behavior.